maintenance/cleanup.py

```diff
@@ -0,0 +1,183 @@
+#   (at your option) any later version.
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+
+"""
+This script first finds and lists, then cleans up objects
+(ROIs, Ratings and Tags) attached to all images contained in a Dataset with a
+specified name belonging to users user-1 through user-40
+and to "trainer-1".
+The listing is done by each user on their own data.
+The information is collected and the final cleanup is done by an
+administrator e.g. "trainer-1", who makes sure that also objects linked to images
+by other users will get unlinked or deleted.
+The cleanup of ROIs and Ratings means deletion of these.
+The cleanup of Tags just deletes the link between the Tag and the image,
+thus removing the Tag, but not deleting it.
+"""
+
+import argparse
+import omero
+from omero.gateway import BlitzGateway
+from omero.rtypes import wrap
+
+obj_ids_taglinks = []
+obj_ids_rating = []
+obj_ids_rois = []
+
+
+def cut_link(conn, username, image):
+    if image is None:
+        print "No image to handle"
+        return
+    params = omero.sys.ParametersI()
+    params.addString('username', username)
+    query = "from Image where name='%s' \
+            AND details.owner.omeName=:username" % image
+    query_service = conn.getQueryService()
+    images = query_service.findAllByQuery(query, params,
+                                          conn.SERVICE_OPTS)
+    if len(images) == 0:
+        print "No image with name %s found" % image
+        return
+    params = omero.sys.ParametersI()
+    params.addLong('imageId', images[0].getId().getValue())
+    query = "from DatasetImageLink where child.id=:imageId"
+    links = conn.getQueryService().findAllByQuery(query, params,
+                                                  conn.SERVICE_OPTS)
+    if len(links) > 0:
+        ids = []
+        for link in links:
+            ids.append(link.getId().getValue())
+        conn.deleteObjects("DatasetImageLink", ids, deleteAnns=True,
+                           deleteChildren=False, wait=True)
+
+
+def list_objs(conn, username, target):
+    if target is None:
+        print "No dataset to handle"
+        return
+    params = omero.sys.ParametersI()
+    params.addString('username', username)
+    query = "from Dataset where name='%s' \
+             AND details.owner.omeName=:username" % target
+    query_service = conn.getQueryService()
+    datasets = query_service.findAllByQuery(query, params, conn.SERVICE_OPTS)
```

Comment:
This query does not seem to find the dataset when the admin is looking for it.
```
PYTHONPATH=$PYTHONPATH:. python ~/Work/training-scripts/maintenance/cleanup.py --server eel.openmicroscopy.org --name root --dataset testing-name $PWD
...
No dataset with name testing-name found
No image to handle
user-4
...
```

But it will find the dataset if ``user-4`` is passed (user-4 is the owner of the dataset, but not the admin which I wanted to do the cleanup as)

```
PYTHONPATH=$PYTHONPATH:. python ~/Work/training-scripts/maintenance/cleanup.py --server eel.openmicroscopy.org --name user-4 --dataset testing-name $PWD

user-3
No dataset with name testing-name found
No image to handle
user-4
No image to handle
user-5

...
```

Reply:
Probably the group context here is not set to -1 or similar